sympy/stats/crv_types.py

```diff
@@ -2195,6 +2196,110 @@ def Normal(name, mean, std):
             MultivariateNormalDistribution, name, mean, std)
     return rv(name, NormalDistribution, (mean, std))
 
+
+#-------------------------------------------------------------------------------
+# Inverse Normal distribution ----------------------------------------------------------
+
+
+class NormalInverseDistribution(SingleContinuousDistribution):
+    _argnames = ('mean', 'shape')
+
+    @property
+    def set(self):
+        return Interval(0, oo)
+
+    @staticmethod
+    def check(mean, shape):
+        _value_check(shape > 0, "Shape parameter must be positive")
```

Comment:
[This comment](https://github.com/sympy/sympy/pull/16576#discussion_r282803523) can be helpful. P

Reply:
I didn't get you. In other distributions also check has been implemented in a similar way. What am I missing here?